hooks/useMqlQuery.ts

```diff
@@ -253,52 +262,64 @@ export default function useMqlQuery({
         handleCombinedError(error);
       }
     });
-  }, [queryInput, metricName, mqlServerUrl]);
-
-  const [{ data, error }, refetchMqlQuery] = useQuery<
-    FetchMqlTimeSeriesQuery,
-    FetchMqlTimeSeriesQueryVariables
-  >({
-    query: FetchMqlQueryTimeSeries,
-    variables: {
-      queryId: state.queryId || "",
-    },
-    pause: !state.queryId || state.cancelledQueries.includes(state.queryId),
-  });
-
-  // I am creating this!
-  useEffect(() => {
-    if (error) {
-      dispatch({
-        type: "fetchResultsFail",
-        errorMessage: getErrorMessage(error),
+  }, [queryInput, metricName, mqlServerUrl, skip]);
+
+  // Note: Extensive effort has gone into using useQuery here rather than an imperative fetch,
```

Comment:
FYI I made this change before I found the _real_ issue here: https://github.com/transform-data/transform-react/pull/7/files#diff-9803adcea4a7f8ac29ceaaa12cef587fe665cc2a4d9cd41d6555a362deb9a0d0L135.

The `getToken` function was different on every render, and this was causing the infinite loop.

I'll take a minute to see if I can unwind this change and get it working the way we want. A consistent approach of using hooks would be more ideal than a mix of imperative and declarative code in here.